gulpfile.js

```diff
@@ -229,25 +252,30 @@ gulp.task('css:watch', () => {
             browserSync.reload();
         }
     });
-});
+
+    cb();
+}
 
 /**
  * Combine the JS files into bundles
  * based on definitions in the `media/static-bundles.json` file.
  */
-gulp.task('js:compile', ['assets'], () => {
+function jsCompileBundles() {
     return merge(staticBundles.js.map(bundleJsFiles));
-});
+}
 
 /**
- * Watch all JS and rebundle as necessary.
+ * Watch for changes to JS files in the `static_final` directory
+ * and recompile bundles when necessary.
  */
-gulp.task('js:watch', () => {
-    return gulp.watch(finalDir + '/js/**/*.js', { debounceDelay: 200 }, event => {
```

Comment:
Note for reviewer: the old `gulp-watch` plugin is now deprecated in favor of the newer built in watch methods. In order to watch for changes on individual files, we now need to use the `on('change')` listener. Doing this means we can no longer use `debounceDelay`. From my testing so far, I haven't noticed any issues in removing it, but I'm not quite clear why it was here originally.

Reply:
It was a way of preventing a cascading rebuild that kinda never stopped. I went through a number of iterations before finding something that worked well, so it's entirely possible that it wasn't actually required in the end and it was another change that fixed it and this was just here superfluously. So if it's working the new way then great 👍